File: app/src/main/java/com/example/moviemvp/MovieList/MovieListContract.java
package com.example.moviemvp.MovieList;

import com.example.moviemvp.Model.Movie;

import java.util.List;

public interface MovieListContract {
    interface Model {
        interface OnFinishedListener {
            void onFinished(List<Movie> movieArrayList);

            void onFailure(Throwable t);
        }

        void getMovieList(OnFinishedListener onFinishedListener, int pageNo);

    }

    interface View {
        void onSuccess();

        void onFailure();

        void setDataToRecyclerView(List<Movie> movieList);

        void onResponseFailure(Throwable t);
    }

    interface Presenter {
        void onDestroy();

        void getMoreData(int pageNo);

        void requestDataFromServer();
    }
}

File: app/src/main/java/com/example/moviemvp/movie_list/MovieListContract.kt
package com.example.moviemvp.movie_list

import com.example.moviemvp.model.Movie

interface MovieListContract {

    interface Model {

        interface OnFinishedListener {

            fun onSuccess(movieArrayList: List<Movie>)

            fun onFailure(t: Throwable)
        }

        fun getMovieList(onFinishedListener: OnFinishedListener, pageNo: Int)

    }

    interface View {

        fun onSuccess(movieList: List<Movie>)

        fun onFailure(t: Throwable)
    }

    interface Presenter {

        fun onDestroy()

        fun requestDataFromServer()
    }
}
